src/Silex/ControllerCollection.php

```diff
@@ -57,11 +59,21 @@ public function __construct(Route $defaultRoute, $routesFactory = null)
     /**
      * Mounts controllers under the given route prefix.
      *
-     * @param string               $prefix      The route prefix
-     * @param ControllerCollection $controllers A ControllerCollection instance
+     * @param string                        $prefix      The route prefix
+     * @param ControllerCollection|callable $controllers A ControllerCollection instance or a callable for defining routes
+     *
+     * @throws \LogicException
      */
-    public function mount($prefix, ControllerCollection $controllers)
+    public function mount($prefix, $controllers)
     {
+        if (is_callable($controllers)) {
+            $collection = $this->controllersFactory ? call_user_func($this->controllersFactory) : new static(new Route(), new RouteCollection());
+            call_user_func($controllers, $collection);
+            $controllers = $collection;
```

Comment:
@ragboyjr sorry to get back to this, but I'm kind of confused about the final implementation.
Little example about what I try to do (always easier then typing lots of text ;) )

```php
// foo/module/events/test
class FooEventsProvider implements \Silex\Api\ControllerProviderInterface
{
    public function connect(\Silex\Application $app)
    {
        $controllers = $app['controllers_factory'];
        $controllers->get('/test', function($id){ return "events ".$id;});
        return $controllers;
    }
}

//foo/module/bars/test
class FooBarsProvider implements \Silex\Api\ControllerProviderInterface
{
    public function connect(\Silex\Application $app)
    {
        $controllers = $app['controllers_factory'];
        $controllers->get('/test', function($id){ return "bars ".$id;});
        return $controllers;
    }
}

$app->mount('/foo/{id}', function ($foo) {
    $foo->mount('/events', new FooEventsProvider());
    $foo->mount('/bars', new FooBarsProvider());
});
```

This won't work because the ControllerCollection cannot handle ControllerProviderInterface instances (for that it needs a ref. to the Application). I think this inline with your first sample which is not implemented (all good).
So switch to the second sample I tried to setup the mounting like this;

```php
$app->mount('/foo/{id}', function ($foo) use ($app) {
    $app->mount($foo, '/events', new FooEventsProvider());
    $app->mount($foo, '/bars', new FooBarsProvider());
});
```

This doesn't work because the Application::mount expects two parameters and not 3. 

Any suggestions how to do this?